### DISCUSSION TITLE
How to add health check for single aws ec2 cluster?

### DISCUSSION THREAD

**Discussion by lukezx3** [milvus-io/milvus#35016] [Q&A and General discussion]
Currently, for aws ec2, milvus only offers health check over gRPC which is built on http2. i am only using a single aws ec2 cluster which relies on http1 and don't want to use a whole k8 setup. how would i implement health checks in my case?

**Comment by xiaofan-luan** [comment]
each node also has it's own check health , which is based on restful

**Reply by lukezx3** [reply]
can you provide more explanation on how to do this? sorry im still a beginner to some of these concepts.

**Comment by xiaofan-luan** [comment]
you can check this test

TestHealthzHandler

Healthz is the health check for each node

**Reply by lukezx3** [reply]
I can’t find any information on it, can you send a link?

**Reply by xiaofan-luan** [reply]
Sorry there is no link, you have to read code and figure out.
We recommend to use docker or K8s to deploy and for single node machine usaully no one do health check. Simply write a daemon thread to bring up the node if crashed

**Reply by lukezx3** [reply]
Yeah but I can’t find any information if milvus health check can be performed on single ec2 cluster (don’t want to use k8 as that’s way too big of a system for my purpose)

**Comment by xiaofan-luan** [comment]
I'm suggesting to use standalone or deploy cluster in K8s. Milvus is burn with K8s, it's gonna to hard to manage the cluster without K8s.

or the easiest way is to buy zilliz cloud(https://zilliz.com/). Both Serverless or Dedicated is easy to scale and deploy.
